.github/custom/clippy.toml

```diff
@@ -0,0 +1,21 @@
+disallowed-from-async-methods = [
+    "tokio::runtime::Handle::block_on",
+    "tokio::runtime::Runtime::block_on",
+    "tokio::task::LocalSet::block_on",
+    "tokio::sync::Mutex::blocking_lock",
+    "tokio::sync::RwLock::blocking_read",
+    "tokio::sync::mpsc::Receiver::blocking_recv",
+    "tokio::sync::mpsc::UnboundedReceiver::blocking_recv",
+    "tokio::sync::oneshot::Receiver::blocking_recv",
+    "tokio::sync::mpsc::Sender::blocking_send",
+    "tokio::sync::RwLock::blocking_write",
+]
+async-wrapper-methods = [
+    "tokio::runtime::Handle::spawn_blocking",
+    "task_executor::TaskExecutor::spawn_blocking",
+    "task_executor::TaskExecutor::spawn_blocking_handle",
+    "warp_utils::task::blocking_task",
+    "warp_utils::task::blocking_json_task",
+    "validator_client::http_api::blocking_signed_json_task",
+    "execution_layer::test_utils::MockServer::new",
```

Comment:
Marking this as an insulating wrapper is a workaround for the clever tricks that `MockServer::new` pulls here:

https://github.com/sigp/lighthouse/blob/cfd26d25e039ed85b67a0b88f02ef633d64b9119/beacon_node/execution_layer/src/test_utils/mod.rs#L82-L90

Clippy just sees that call to `block_on` and thinks that it's dodgy. It isn't smart enough to work out whether it's reachable at runtime (and I think that's probably undecidable :grin:).

We may want to use this technique of manually marking a function as a wrapper in other places. It tells Clippy "don't check the functions called by this function, I've checked them myself and deemed them to be safe". We should use this sparingly.